apps/wallet-mobile/src/yoroi-wallets/cardano/byron/ByronWallet.ts

```diff
@@ -736,7 +743,9 @@ export class ByronWallet implements YoroiWallet {
         ? [stakingPrivateKey]
         : undefined
 
-    if (datum) {
+    const datum = unsignedTx.recipientEntries.find((entry) => entry.datum)?.datum
```

Comment:
it doesn't need to find the first, just build an array with all datum if there is datum, filter by contains datum, the signTx, handles the array of datum to build the script_data_hash

Reply:
if it's possible to update the sign, it can read from UnsingTx.outputs.datum, it looks wrong to pass it again, once it is already there.